frontend/packages/process-editor/src/components/ConfigPanel/ConfigContent/EditActions/EditActions.test.tsx

```diff
@@ -1,148 +1,161 @@
+    setupBpmnActionModelerMock({
+      addNewActionToTaskMock,
+      updateTypeForActionMock,
+      updateActionNameOnActionElementMock,
     });
-    const addNewActionButton = screen.getAllByRole('button', {
+
+    renderEditActions();
+    const addButton = screen.getByRole('button', {
       name: textMock('process_editor.configuration_panel_actions_add_new'),
     });
-    expect(addNewActionButton).toHaveLength(1);
-  });
 
-  it('should render existing actions when task have predefined actions', () => {
-    const predefinedAction =
-      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[0].action;
-    const customServerAction =
-      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[1].action;
-    const customProcessAction =
-      mockBpmnDetails.element.businessObject.extensionElements.values[0].actions.action[2].action;
-    renderEditActions();
+    await user.click(addButton);
+    await waitFor(() => expect(addNewActionToTaskMock).toHaveBeenCalledTimes(1));
     expect(
-      screen.getByRole('button', {
-        name: textMock('process_editor.configuration_panel_actions_action_label', {
+      screen.getByText(
+        textMock('process_editor.configuration_panel_actions_action_card_title', {
           actionIndex: 1,
-          actionName: predefinedAction,
-        }),
-      }),
-    ).toBeInTheDocument();
-    expect(
-      screen.getByRole('button', {
-        name: textMock('process_editor.configuration_panel_actions_action_label', {
-          actionIndex: 2,
-          actionName: customServerAction,
         }),
-      }),
-    ).toBeInTheDocument();
-    expect(
-      screen.getByRole('button', {
-        name: textMock('process_editor.configuration_panel_actions_action_label', {
-          actionIndex: 3,
-          actionName: customProcessAction,
-        }),
-      }),
-    ).toBeInTheDocument();
+      ),
+    );
   });
 
-  it.each(['confirmation', 'signing', 'payment'])(
-    'should render readOnly non-clickable defined action button for actions that are required for task type: %s',
-    (taskType: BpmnTaskType) => {
-      const actions = actionsForTaskTypes[taskType];
-      const element = { businessObject: { extensionElements: { values: [actions] } } };
-      renderEditActions({ ...mockBpmnDetails, taskType: taskType, element });
-      actions.actions.action.forEach((action, index) => {
-        const definedActionButton = screen.getByRole('button', {
-          name: textMock('process_editor.configuration_panel_actions_action_label', {
-            actionIndex: index + 1,
-            actionName: action.action,
-          }),
-        });
-        expect(definedActionButton).toHaveAttribute('aria-readonly');
-      });
-    },
-  );
+  it('should append new action if actions already exists', async () => {
+    const user = userEvent.setup();
+
+    const addNewActionToTaskMock = jest.fn();
```

Comment:
Maybe we should move these functions to the top of the file, and then they can be used both in this test, and the test below? 
I can see that the names are the same, but the way they are used are different 😄

Reply:
I have by design avoided placing them at the top of the file. I want to isolate each `it` as much as possible. Adding method mocks at the top would mean sharing the same reference between tests. By duplicating them within each `it`, it is very clear which methods you are actually testing within a specific `it`. You can easily see which mocks are necessary for the test to run. This approach also aligns with Testing Library’s recommendations to keep each `it/test` as isolated as possible from one another.

Reply:
Great explanation! 😄 
I believe this is something we should bring up with the team as I have seen a lot of people doing this differently. We should aim to the same approach for all our tests to ensure that we have equal ways of doing things 🙌

Reply:
I have created a meeting next week to talk about testing. 😊